examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -158,10 +164,21 @@ CHIP_ERROR Commands::RunCommand(NodeId localId, NodeId remoteId, int argc, char
         execContext.commissioner  = &mController;
         execContext.opCredsIssuer = &mOpCredsIssuer;
         execContext.storage       = &mStorage;
+        execContext.localId       = localId;
+        execContext.remoteId      = remoteId;
```

Comment:
While it is reasonable ok to stash away localId and remoteId as part of the execution context, I presume you also did this to avoid needing to pass in a richer context to RunQueuedCommand, given its limited function signature.

Would be worthwhile exploring a common `PostClosureAsync(std::function<void()>)` method on both the POSIX and Darwin implementations to permit a bit more flexibility. This isn't needed in this PR, but just a thought.

Reply:
I most definitely did this to avoid needing to pass a richer context to RunQueuedCommand.

And yes, I agree that having a way to post a nicer runnable that can more easily close over things to the chip event loop would be nice.